### PR TITLE
[BugFix] Support list-based boolean masks for TensorDict

### PR DESCRIPTION
**Pull request by benoitdescamps**
## Description

Fixes issue https://github.com/facebookresearch/rl/issues/258

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/facebookresearch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by vmoens** [comment]
Thanks for picking this up @benoitdescamps 
I'm looking into the failing test. 
There is a bunch of stuff that needs to be fixed for this PR to be mergeable, I'm taking care of it. Bear with me, I'll do a PR on your branch as soon as I'm done with it!

**Comment by vmoens** [comment]
We have device issues in the tests, do you think you can handle it @benoitdescamps ?

**Comment by benoitdescamps** [comment]
> We have device issues in the tests, do you think you can handle it @benoitdescamps ?

yes, I will have a go at it

File: test/test_tensordict.py [hunk 1]
```diff
@@ -198,9 +198,19 @@ def test_mask_td(device):
         "key2": torch.randn(4, 5, 10, device=device),
     }
     mask = torch.zeros(4, 5, dtype=torch.bool, device=device).bernoulli_()
+    mask_list = mask.numpy().tolist()
```
Comment (vmoens) [inline test/test_tensordict.py:201]: this should be moved to cpu too

Comment (benoitdescamps) [reply test/test_tensordict.py]: done

**Review by vmoens — APPROVED** [review]
LGTM
Thanks for this feature! Also helped to uncover a couple of bugs, which is great!
